File: Eris./Models/Message.swift
//
//  Message.swift
//  Eris.
//
//  Created by Ignacio Palacio on 19/6/25.
//

import Foundation
import SwiftData

@Model
class Message: Identifiable {
    var id: UUID
    var content: String
    var role: Role
    var timestamp: Date
    var thread: Thread?
    
    enum Role: String, Codable {
        case user
        case assistant
        case system
    }
    
    init(content: String, role: Role) {
        self.id = UUID()
        self.content = content
        self.role = role
        self.timestamp = Date()
    }
}

File: app/src/main/java/dev/natxo/eris/data/database/entities/Message.kt
package dev.natxo.eris.data.database.entities

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

@Entity(
    tableName = "messages",
    foreignKeys = [
        ForeignKey(
            entity = Thread::class,
            parentColumns = ["id"],
            childColumns = ["threadId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Message(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),
    val threadId: String,
    val content: String,
    val role: MessageRole,
    val timestamp: Date = Date()
)

enum class MessageRole {
    USER, ASSISTANT, SYSTEM
}